include/certifier_framework.i

```diff
@@ -13,9 +13,34 @@
 %module certifier_framework
 %include "std_string.i"
 
-// Needed to invoke init_policy_key() using Python bytestream arg.
+// Needed to invoke init_policy_key() and other interfaces that take
+// (byte *, int) supplied by Python bytestream arg.
 %include <pybuffer.i>
-%pybuffer_binary(byte* asn1_cert, int asn1_cert_size)   // cc_trust_data()->init_policy_key()
+
+// These mappings are required for invoking python_initialize_simulated_enclave()
+// through Python apps
+%pybuffer_binary(const byte *serialized_attest_key, int attest_key_size);
+%pybuffer_binary(const byte *measurement, int measurement_size);
+%pybuffer_binary(const byte *serialized_attest_endorsement, int attest_key_signed_claim_size);
+
+%pybuffer_binary(byte* asn1_cert, int asn1_cert_size);   // cc_trust_data()->init_policy_key()
+
+// ----------------------------------------------------------------------------
+// Ignore few overloaded interfaces so that there is no need to exercise the
+// disambiguation code in SWIG-generated wrapper module(s). Otherwise,
+// while trying to select the right interface to invoke based on the
+// arguments supplied, we run afoul of the limitation registered under
+// SWIG issue https://github.com/swig/swig/issues/1916
+// ----------------------------------------------------------------------------
+%ignore initialize_simulated_enclave(const string &serialized_attest_key,
+                                     const string &measurement,
+                                     const string &serialized_attest_endorsement);
+
+%ignore init_client_ssl(const string &host_name, int port,
+                        const cc_trust_manager &mgr);
+
+%ignore init_server_ssl(const string &host_name, int port,
+                        const cc_trust_manager &mgr);
```

Comment:
You sure we could ignore these for the Python simple app?

Reply:
Hmmm ... Thanks for spotting this @yelvmw -- This made me do some more thinking and debugging / testing investigation.

I made the following changes in response to your comment:

- These two `ignore` directives are not needed for tests in `test_certifier_framework.py` to run cleanly.
- So commented them out, but left them behind. Added an elaborate comment explaining when / why we might need these to be re-enabled. (For future reference; I'm going to forget all this stuff ...)
- Added couple of new test cases to verify that these interfaces can still be invoked with `const cc_trust_manager &mgr` argument . And that the tests will still succeed. 
   - Add test case `test_run_app_as_a_client_init_client_ssl_with_trust_manager()`
   - Add test case `test_run_app_as_a_server_with_trust_manager()`
 
- Figured the an alternate fix to avoid the 120 seconds sleep in some test cases. In the method `server_dispatch()`, for the testing-hook when call-back function hdlr is NULL, also do `close((sock);` before exiting. This should take away the unpredictability seen in CI runs. (Will re-test and verify.)